framework/auth/core.py

```diff
@@ -1381,13 +1385,19 @@ def is_affiliated_with_institution(self, inst):
         return inst in self.affiliated_institutions
 
     def remove_institution(self, inst_id):
-        try:
-            self.affiliated_institutions.remove(inst_id)
-        except ValueError:
-            return False
-        return True
+        removed = False
+        for inst in self.affiliated_institutions:
+            if inst._id == inst_id:
+                self.affiliated_institutions.remove(inst)
+                removed = True
+        return removed
+
+    _affiliated_institutions = fields.ForeignField('node', list=True)
 
-    affiliated_institutions = fields.ForeignField('institution', list=True)
+    @property
+    def affiliated_institutions(self):
+        from website.project.model import Institution, SpecialList
+        return SpecialList(self, '_affiliated_institutions', [Institution(inst) for inst in self._affiliated_institutions])
```

Comment:
`SpecialList` is not a descriptive name. How about `ProxiedList`, since `affiliated_institutions` acts as a proxy for `_affiliated_institutions`?